matlab/CARFAC_AGC_Step.m

```diff
@@ -19,11 +19,31 @@
 function [state, updated] = CARFAC_AGC_Step(detects, coeffs, state)
 % function [state, updated] = CARFAC_AGC_Step(detects, coeffs, state)
 %
-% one time step of the AGC state update; decimates internally
+% one time step of the AGC state update; maybe decimates internally.
+% AGC input is detects; detect_scale is already in the coefficients.
 
-stage = 1;
-AGC_in = coeffs(1).detect_scale * detects;
-[state, updated] = CARFAC_AGC_Recurse(coeffs, AGC_in, stage, state);
+if coeffs.non_decimating  
```

Comment:
We should add a test to ensure that the non-decimating code path behaves the same as the decimating code path with all decimations equal to 1.

Reply:
It was actually my intention that they be a little different, with more delay from later stages in the non-decimating version to make it closer to the inherent delay when decimating; the decimating code evaluation order tries to minimize delay while the non-decimating maximizes it, for a closer match to the decimating case.  For the steady-state response, though, they should match very nearly (subject to some ripple when the final state is sampled).  So maybe I'll add a test for that by setting non_decimating to false in the AGC_Coeffs in a test with [1, 1, 1, 1].